internal/tokenizer/tokenize_test.go

```diff
@@ -89,27 +90,45 @@ var (
 		"XHTML", "sample", "file", "type", "#example", "background", "color", "yellow", "id", "Just", "a", "simple", "XHTML", "test", "page.",
 		"-", "|", "+", "&&", "<", "<", "-", "!", "!", "!", "=", "=", "!", ":", "=", ":", "=", ",", ",", "=", ">", ">", "=", "=", "=", "=", ">",
 		"'", ",", ">", "=", ">", "=", "=", ">", "=", ">", ":", ">", "=", ">"}
-)
 
-func TestTokenize(t *testing.T) {
-	tests := []struct {
+	Tests = []struct {
 		name     string
 		content  []byte
 		expected []string
 	}{
 		{name: "content", content: []byte(testContent), expected: tokensFromTestContent},
 	}
+)
 
-	for _, test := range tests {
+func TestTokenize(t *testing.T) {
+	for _, test := range Tests {
 		t.Run(test.name, func(t *testing.T) {
 			before := string(test.content)
 			tokens := Tokenize(test.content)
 			after := string(test.content)
-			assert.Equal(t, before, after, "the input slice was modified")
-			assert.Equal(t, len(test.expected), len(tokens), fmt.Sprintf("token' slice length = %v, want %v", len(test.expected), len(tokens)))
+			require.Equal(t, before, after, "the input slice was modified")
+			require.Equal(t, len(test.expected), len(tokens), fmt.Sprintf("token' slice length = %v, want %v", len(test.expected), len(tokens)))
 			for i, expectedToken := range test.expected {
 				assert.Equal(t, expectedToken, tokens[i], fmt.Sprintf("token = %v, want %v", tokens[i], expectedToken))
 			}
 		})
 	}
 }
+
+func BenchmarkTokenizer_BaselineCopy(b *testing.B) {
+	b.ReportAllocs()
+	for i := 0; i < b.N; i++ {
+		for _, test := range Tests {
+			test.content = append([]byte(nil), test.content...)
```

Comment:
Is there any reason to replace `test.content` with the result here? 

I worry that we're mostly testing the speed of the memory allocator here. If we really care only about copy as a baseline, I suggest pre-allocating:

```go
var size int
for _, test := range Tests {
   if len(test.content) > size {
      size = len(test.content)
   }
}
buf := make([]byte, 0, size)
for _, test := range Tests {
   _ = append(buf[:0], test.content...)
}
```

Reply:
Also note that we are capping Tokenize calls at `byteLimit`, but not doing so for our baseline copies. I think we should probably do the same in both cases for a fair baseline.

Reply:
> I worry that we're mostly testing the speed of the memory allocator here

sorry for may be a miss-leading name of the test, but including allocation is expected in this case - it mimics the behaviour of current tokeniser, [up to the point of tokenisation](https://github.com/src-d/enry/blob/ab3c26b46dfaf8281bf78f2e998111e6a40829bf/internal/tokenizer/tokenize.go#L22).

But also it exists just to provide a feeling of an order of magnitude of time that it takes of doing everything, but actual tokenisation, rather than some fine-grained apple-to-apple comparison.

> Is there any reason to replace test.content with the result here?
Not really, will update